compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -746,8 +746,17 @@ impl<'tcx> TypeVisitor<'tcx> for OrphanChecker<'tcx> {
         result
     }
 
-    // FIXME: Constants should participate in orphan checking.
     fn visit_const(&mut self, _c: ty::Const<'tcx>) -> ControlFlow<Self::BreakTy> {
+        // All possible values for a constant parameter already exist
+        // in the crate defining the trait, so they are always non-local.
```

Comment:
Hmm is `const Foo: &dyn Bar` ever even possible? What about `<T: Trait, const Foo: T>`?

I think this statement holds today, but if it could ever change in the future, do we have some way to ensure we don't forget about this?

Reply:
:thinking: 

so `T: Trait, const VALUE: T` isn't an issue, because `T` has to be forward declared, so if `T` is a local type, then the orphan check already succeeds the moment it sees `T`

for trait objects that's more difficult :grin: or well, it depends on whether we already have local values for `const T: fn()`. going to extend that comment.

feel like the value of allowing uncovered const params is still greater then allowing impls to be guarded by a local function pointer :grin: so this impl is still the right one imo